pkg/provider/core.go

```diff
@@ -226,23 +227,24 @@ func (p *Provider) GetMachineStatus(ctx context.Context, req *driver.GetMachineS
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
-	klog.V(2).Infof("machine get request has been processed successfully for %q", req.Machine.Name)
-
 	if resp.Machine.Allocation == nil {
+		klog.V(2).Infof("machine already released: %q", req.Machine.Name)
 		return nil, status.Error(codes.NotFound, "machine already released")
```

Comment:
If the machine is already deleted, shouldnt we return a nil error because otherwise the reconcile will never end ?

Reply:
It says in the function docstring:
```
// The request should return a NOT_FOUND (5) status error code if the machine is not existing
```

Reply:
Ah, i see